pcdsdevices/pneumatic.py

```diff
@@ -0,0 +1,69 @@
+
+"""
+Pneumatic Classes.
+
+This Module contains all the classes relating to Pneumatic Actuators
+"""
+
+from lightpath import LightpathState
+from ophyd import Component as Cpt
+from ophyd import EpicsSignal, EpicsSignalRO
+
+from pcdsdevices.interface import LightpathMixin
+
+
+class BeckhoffPneumatic(LightpathMixin):
+    """
+    Class containing basic Beckhoff Pneumatic support
+    """
+    lightpath_cpts = ['limit_switch_in', 'limit_switch_out']
+
+    # readouts
+    limit_switch_in = Cpt(EpicsSignalRO, ':PLC:bInLimitSwitch')
+    limit_switch_out = Cpt(EpicsSignalRO, ':PLC:bOutLimitSwitch')
+
+    retract_status = Cpt(EpicsSignalRO, ':bRetractDigitalOutput')
+    insert_status = Cpt(EpicsSignalRO, ':bInsertDigitalOutput')
+
+    # logic and supervisory
+    interlock_ok = Cpt(EpicsSignalRO, 'bInterlockOK')
+    insert_ok = Cpt(EpicsSignalRO, 'bInsertEnable')
+    retract_ok = Cpt(EpicsSignalRO, 'bretractEnable')
+
+    # commands
+    insert_signal = Cpt(EpicsSignal, 'CMD:IN')
+    retract_signal = Cpt(EpicsSignal, 'CMD:OUT')
+
+    # returns
+    busy = Cpt(EpicsSignalRO, ':bBusy')
+    done = Cpt(EpicsSignalRO, ':bDone')
+    reset = Cpt(EpicsSignal, ':bReset')
+    error = Cpt(EpicsSignalRO, ':PLC:bError')
+    error_id = Cpt(EpicsSignalRO, ':PLC:nErrorId')
+    error_message = Cpt(EpicsSignalRO, ':PLC:sErrorMessage')
```

Comment:
It's not really needed for today, but you could imagine some future version of this that waits for the busy/done signals to finish when you insert or retract and reports the error_message from the PLC

Reply:
Would this be blocking? or some sort of thread code that code goes to sleep?

Reply:
it should be pretty quick to write a loop with a time limit to do this.

Reply:
implemented a loop with a timeout. wdy think?

Reply:
A full implementation would have something like
```
def insert(self, wait: bool = False, timeout: float = 10.0) -> Status:
    ...
```
And then you'd use the ophyd status objects to communicate the error states.

The loop is solid but:
- you'd want it to be optional
- re-use ophyd move kwargs like `wait` and `timeout` for familiarity
- should be a subroutine that's used in both insert and remove

Reply:
Okay, let me take a look at these suggestions. Thank you.

Reply:
I re-implemented this using status. Will port this so a callable routine so remove and also use it if this looks good.